GoogleForCreators/web-stories-wp
# Enable smart styling for presets as opt-in feature

## Summary
- Removes the smart color feature from the quick-add Text fully
- Removes the Presets header (see [Slack discussion](https://xwp.slack.com/archives/C0135V8LYJE/p1632746409059200?thread_ts=1632324004.044800&cid=C0135V8LYJE))
- Adds a Toggle for text presets and text sets that turns the smart color feature on/off
- Adds tooltip that warns about performance when toggling.


## Relevant Technical Choices
- Removes the feature flags fully since if needed we can quickly turn these features off by just hiding the toggle button.


## To-do

- [x] Remove the smart color from Text quick-action fully?
- [x] Confirm the Toggle for text presets vs Text Sets + re-position as needed
- [x] Enable for Text Sets as well? TBC
- [ ] Verify tests



## User-facing changes
<img width="348" alt="Screenshot 2021-09-27 at 16 34 49" src="https://user-images.githubusercontent.com/3294597/134918979-3fcdd39d-ea1f-413d-9df4-ed60e58ee28f.png">


## Testing Instructions





This PR can be tested by following these steps:

1. Switch to the Text tab
2. Toggle On "Use auto styled text" 
3. Add some background (e.g. black) that would trigger adding auto-style
4. Test with both Texts and Text Sets to verify that when auto styling is turned on, it works. (Note: it's not expected to work when dragging and dropping, only when clicking)
5. Now toggle off auto styling.
6. Verify the text isn't auto-styled anymore when adding.
7. Verify that clicking the `+` sign in the Text tab directly (the quick-action for adding a text) does not apply auto-color even if the toggle is on.


## Checklist




---



Fixes #8238 
Fixes #8740
